library/spdm_responder_lib/measurements.c

```diff
@@ -246,266 +247,181 @@ return_status spdm_get_response_measurements(IN void *context,
-					return RETURN_SUCCESS;
-				}
-				spdm_response->header.param2 = slot_id_param;
-			}
-		} else {
-			spdm_create_measurement_opaque(spdm_context,
-						       spdm_response,
-						       spdm_response_size);
-		}
 		break;
 
 	default:
-		measurment_record_size = 0;
-		cached_measurment_block = (void *)device_measurement;
-		for (index = 0; index < device_measurement_count;
-				index++) {
-			measurment_block_size =
-				sizeof(spdm_measurement_block_dmtf_t) +
-				cached_measurment_block
-					->Measurement_block_dmtf_header
-					.dmtf_spec_measurement_value_size;
-			if (cached_measurment_block->Measurement_block_common_header.index == 
-				spdm_request->header.param2) {
-				measurment_record_size =
-					measurment_block_size;
-				break;
-			}
-			cached_measurment_block =
-				(void *)((uintn)cached_measurment_block +
-						measurment_block_size);
-		}
-		if (index != device_measurement_count ) {
-			spdm_response_size =
-				sizeof(spdm_measurements_response_t) +
-				measurment_record_size;
-			if ((spdm_request->header.param1 &
-			     SPDM_GET_MEASUREMENTS_REQUEST_ATTRIBUTES_GENERATE_SIGNATURE) !=
-			    0) {
-				spdm_response_size += measurment_sig_size;
-			} else {
-				spdm_response_size += measurment_no_sig_size;
-			}
 
-			ASSERT(*response_size >= spdm_response_size);
-			*response_size = spdm_response_size;
-			zero_mem(response, *response_size);
-			spdm_response = response;
-
-			if (spdm_is_version_supported(
-				    spdm_context, SPDM_MESSAGE_VERSION_11)) {
-				spdm_response->header.spdm_version =
-					SPDM_MESSAGE_VERSION_11;
-			} else {
-				spdm_response->header.spdm_version =
-					SPDM_MESSAGE_VERSION_10;
-			}
-			spdm_response->header.request_response_code =
-				SPDM_MEASUREMENTS;
-			spdm_response->header.param1 = 0;
-			spdm_response->header.param2 = 0;
-			spdm_response->number_of_blocks = 1;
-			spdm_write_uint24(
-				spdm_response->measurement_record_length,
-				(uint32)measurment_record_size);
-
-			measurment_block = (void *)(spdm_response + 1);
-			copy_mem(measurment_block,
-					cached_measurment_block,
-					measurment_block_size);
-
-			if ((spdm_request->header.param1 &
-			     SPDM_GET_MEASUREMENTS_REQUEST_ATTRIBUTES_GENERATE_SIGNATURE) !=
-			    0) {
-				if (spdm_response->header.spdm_version >=
-				    SPDM_MESSAGE_VERSION_11) {
-					slot_id_param =
-						spdm_request->SlotIDParam;
-					if ((slot_id_param != 0xF) &&
-					    (slot_id_param >=
-					     spdm_context->local_context
-						     .slot_count)) {
-						spdm_generate_error_response(
-							spdm_context,
-							SPDM_ERROR_CODE_INVALID_REQUEST,
-							0, response_size,
-							response);
-						return RETURN_SUCCESS;
-					}
-					spdm_response->header.param2 =
-						slot_id_param;
-				}
-			} else {
-				spdm_create_measurement_opaque(
-					spdm_context, spdm_response,
-					spdm_response_size);
-			}
-		} else {
-			//Block not found
+		if (measurements_count != 1) { //Block not found

```

Comment:
This is the case where spdm_measurement_collection returns TRUE but the measurement_index was not valid for the device.  We return SPDM_ERROR_CODE_INVALID_REQUEST instead of SPDM_ERROR_CODE_UNSPECIFIED.

Reply:
Agree. Should probably assert measurements count == 1 after this section too.